src/spawn.cpp

```diff
@@ -198,14 +200,18 @@ bool Spawn::isInSpawnZone(const Position& pos)
 bool Spawn::spawnMonster(uint32_t spawnId, MonsterType* mType, const Position& pos, Direction dir, bool startup /*= false*/)
 {
 	std::unique_ptr<Monster> monster_ptr(new Monster(mType));
-	if (startup) {
-		//No need to send out events to the surrounding since there is no one out there to listen!
-		if (!g_game.internalPlaceCreature(monster_ptr.get(), pos, true)) {
-			return false;
-		}
+	if (!g_events->eventMonsterOnSpawn(monster_ptr.get(), pos, startup, false)) {
+		return false;
 	} else {
```

Comment:
The `else` is not necessary. You can reduce the level of indentation here since you are returning early.

Reply:
> You can reduce the level of indentation here since you are returning early.

Not sure what you mean by that.
changes made.